cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -173,6 +177,39 @@ def transcript_download_handler(request):
     return response
 
 
+def upload_transcript(request):
+    edx_video_id = request.POST['edx_video_id']
+    language_code = request.POST['language_code']
+    new_language_code = request.POST['new_language_code']
+    transcript_file = request.FILES['file']
+    try:
+        # Convert SRT transcript into an SJSON format
+        # and upload it to S3.
+        sjson_subs = Transcript.convert(
+            content=transcript_file.read().decode('utf-8'),
+            input_format=Transcript.SRT,
+            output_format=Transcript.SJSON
+        ).encode()
+        create_or_update_video_transcript(
+            video_id=edx_video_id,
+            language_code=language_code,
+            metadata={
+                'provider': TranscriptProvider.CUSTOM,
+                'file_format': Transcript.SJSON,
+                'language_code': new_language_code
+            },
+            file_data=ContentFile(sjson_subs),
+        )
+        response = JsonResponse(status=201)
```

Comment:
Could we please add logging statement for the success case? It really helps monitoring

Reply:
I added a message here as well, also including the edX video ID and transcript language.